citadel_workspace_service/src/kernel/mod.rs

```diff
@@ -178,15 +181,23 @@ async fn payload_handler(
                 None => info!(target: "citadel","connection not found"),
             };
         }
-        InternalServicePayload::ConnectionFailure { .. } => {}
-        InternalServicePayload::MessageReceived { .. } => {}
-        InternalServicePayload::Disconnect { .. } => {}
+
+        InternalServicePayload::Disconnect { cid, uuid } => {
+            let request = NodeRequest::DisconnectFromHypernode(DisconnectFromHypernode {
+                implicated_cid: cid,
+                v_conn_type: VirtualTargetType::LocalGroupServer(cid),
+            });
+            connection_map.remove(&cid);
+            hm.clone().lock().await.remove(&uuid);
```

Comment:
Almost. Just because we're disconnecting from the server does not mean we should disconnect from the TCP connection.
```suggestion
            
```